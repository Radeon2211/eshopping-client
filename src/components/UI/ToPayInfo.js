import React from 'react';
import PropTypes from 'prop-types';
import FlexWrapper from './FlexWrapper';
import PlainText from './PlainText';
import { formatPrice } from '../../shared/utility';

const ToPayInfo = (props) => {
  const { value } = props;

  return (
    <FlexWrapper align="center" justify="center" wrap="wrap" spacing="1">
      <PlainText size="4">To pay</PlainText>
      <PlainText size="6" spacing="1px">
        {formatPrice(value)}
      </PlainText>
    </FlexWrapper>
  );
};

ToPayInfo.propTypes = {
  value: PropTypes.number.isRequired,
};

export default ToPayInfo;
