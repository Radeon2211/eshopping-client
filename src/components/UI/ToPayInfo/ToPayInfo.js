import PropTypes from 'prop-types';
import FlexWrapper from '../FlexWrapper';
import PlainText from '../PlainText';
import { formatPrice } from '../../../shared/utility/utility';

export default function ToPayInfo({ value }) {
  return (
    <FlexWrapper
      $align="center"
      $justify="center"
      $wrap="wrap"
      $spacing="1"
      data-testid="ToPayInfo"
    >
      <PlainText $size="4">To pay</PlainText>
      <PlainText $size="6" $spacing="1px">
        {formatPrice(value)}
      </PlainText>
    </FlexWrapper>
  );
}

ToPayInfo.propTypes = {
  value: PropTypes.number.isRequired,
};
