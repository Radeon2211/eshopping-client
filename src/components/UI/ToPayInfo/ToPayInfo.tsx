import PropTypes from 'prop-types';
import FlexWrapper from '../FlexWrapper';
import PlainText from '../PlainText';
import { formatPrice } from '../../../shared/utility/utility';

interface ToPayInfoProps {
  value: number;
}

export default function ToPayInfo({ value }: ToPayInfoProps) {
  return (
    <FlexWrapper
      $align="center"
      $justify="center"
      $wrap="wrap"
      $spacing="level1"
      data-testid="ToPayInfo"
    >
      <PlainText $size="level4">To pay</PlainText>
      <PlainText $size="level6" $spacing="1px">
        {formatPrice(value)}
      </PlainText>
    </FlexWrapper>
  );
}

ToPayInfo.propTypes = {
  value: PropTypes.number.isRequired,
};
