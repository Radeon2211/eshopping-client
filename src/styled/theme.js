const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#666',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#00A790',
    greenDark: '#009680',
    red: '#F22613',
  },
  shadows: {
    level1: '0 0 2px rgba(0, 0, 0, .1)',
    level2: '0 0 2px rgba(0, 0, 0, .3)',
    level3: '0 0 3px rgba(0, 0, 0, .3)',
    level4: '0 0 5px rgba(0, 0, 0, .3)',
    level5: '0 0 8px rgba(0, 0, 0, .3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Ubuntu, sans-serif',
    text: 'Lato, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
  },
};

export default theme;
