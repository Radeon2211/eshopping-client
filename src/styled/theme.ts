const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#777',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#06AD96',
    greenDark: '#009680',
    red: '#F22613',
    redLight: '#F94D2A',
    redDark: '#985A5A',
  },
  shadows: {
    level1: '0 0 3px rgba(0, 0, 0, 0.03)',
    level2: '0 0 3px rgba(0, 0, 0, 0.3)',
    level3: '0 0 5px rgba(0, 0, 0, 0.3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Noto Sans, sans-serif',
    text: 'Nunito, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
    level5: 100000,
  },
  fontSizes: {
    level1: '1.3rem',
    level2: '1.4rem',
    level3: '1.5rem',
    level4: '1.7rem',
    level5: '2.2rem',
    level6: '2.6rem',
  },
  durations: {
    level1: 0.1,
    level2: 0.2,
    level3: 0.3,
  },
  lineHeights: {
    level1: '1',
    level2: '1.1',
    level3: '1.2',
    level4: '1.3',
    level5: '1.4',
    level6: '1.5',
  },
};

export type ThemeSpacing = keyof typeof theme.spacings;
export type ThemeZIndex = keyof typeof theme.zIndexes;
export type ThemeFontSize = keyof typeof theme.fontSizes;
export type ThemeDuration = keyof typeof theme.durations;
export type ThemeLineHeight = keyof typeof theme.lineHeights;

export default theme;
