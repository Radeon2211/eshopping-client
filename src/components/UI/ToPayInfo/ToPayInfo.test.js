import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ThemeProvider } from 'styled-components';
import ToPayInfo from './ToPayInfo';
import theme from '../../../styled/theme';

const setUp = (value) => {
  return render(
    <ThemeProvider theme={theme}>
      <ToPayInfo value={value} />
    </ThemeProvider>,
  );
};

afterEach(cleanup);

describe('<ToPayInfo />', () => {
  it('should render everything correctly', () => {
    setUp(1195.2);
    expect(screen.getByTestId('ToPayInfo')).toHaveTextContent('To pay');
    expect(screen.getByTestId('ToPayInfo')).toHaveTextContent('$1,195.20');
  });
});
