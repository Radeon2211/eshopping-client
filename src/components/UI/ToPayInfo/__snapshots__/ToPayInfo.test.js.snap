// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToPayInfo /> should render everything correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ hyiZji"
    direction="row"
    spacing="1"
    width="100%"
    wrap="wrap"
  >
    <span
      class="sc-gsTCUz jhJJLb"
      color=""
      display="inline-block"
      overflow="visible"
      size="4"
      spacing="0px"
      transform=""
    >
      To pay
    </span>
    <span
      class="sc-gsTCUz byQcZA"
      color=""
      display="inline-block"
      overflow="visible"
      size="6"
      spacing="1px"
      transform=""
    >
      $1,195.20
    </span>
  </div>
</DocumentFragment>
`;
